apps/explorer/src/hooks/useTokenList.ts

```diff
@@ -0,0 +1,54 @@
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import useSWR from 'swr'
+import { SWR_NO_REFRESH_OPTIONS } from '@cowprotocol/common-const'
+import type { TokenInfo, TokenList } from '@uniswap/token-lists'
+
+type TokenListByAddress = Record<string, TokenInfo>
+type TokenListPerNetwork = Record<SupportedChainId, TokenListByAddress>
+
+const INITIAL_TOKEN_LIST_PER_NETWORK: TokenListPerNetwork = {
+  [SupportedChainId.MAINNET]: {},
+  [SupportedChainId.GNOSIS_CHAIN]: {},
+  [SupportedChainId.SEPOLIA]: {},
+}
+
+export function useTokenList(chainId: SupportedChainId | undefined): { data: TokenListByAddress; isLoading: boolean } {
+  const { data: cowSwapList, isLoading: isCowListLoading } = useTokenListByUrl(
+    'https://files.cow.fi/tokens/CowSwap.json'
+  )
+  const { data: coingeckoList, isLoading: isCoingeckoListLoading } = useTokenListByUrl(
+    'https://tokens.coingecko.com/uniswap/all.json'
```

Comment:
This one is huge (5130 tokens), 292kb size.

Maybe it's better to use lists which we use in CoWSwap by default?

https://tokenlists.org/token-list?url=https://files.cow.fi/tokens/CowSwap.json
https://tokenlists.org/token-list?url=https://tokens.honeyswap.org
https://tokenlists.org/token-list?url=https://files.cow.fi/tokens/CoinGecko.json

Reply:
That's the point!
We want to have as most as possible to load it once, cache it indefinitely and reach onchain sources as little as possible.
This should have minimal performance impact as we keep the list in memory only to load the token info rather than loading token balances.